packages/core/src/auth/sso/ssoAccessTokenProvider.ts

```diff
@@ -199,16 +199,25 @@ export class SsoAccessTokenProvider {
             return await pollForTokenWithProgress(() => this.oidc.createToken(tokenRequest), authorization)
         }
 
-        let token: ReturnType<typeof openBrowserAndWaitUntilComplete>
-        if (telemetry.spans.map(s => s.name).includes('aws_loginWithBrowser')) {
+        const token = this.withBrowserLoginTelemetry(() => openBrowserAndWaitUntilComplete())
+
+        return this.formatToken(await token, registration)
+    }
+
+    /**
+     * Wraps the given function with telemetry related to the browser login.
+     */
+    private withBrowserLoginTelemetry<T extends (...args: any[]) => any>(func: T): ReturnType<T> {
+        if (telemetry.spans.some(s => s.name === 'aws_loginWithBrowser')) {
```

Comment:
If this is true, should we still call

    telemetry.record({ credentialStartUrl: this.profile.startUrl })

? That will update/ovewrite the `credentialStartUrl` field for the current span(s).

Reply:
We're making the assumption that the parent `aws_loginWithBrowser` span (in `auth.ts`) has already set it.

I wanted to keep all values set in the parent one unless we had a reason not to.